pkg/trace/api/otlp.go

```diff
@@ -187,6 +193,108 @@ func (o *OTLPReceiver) SetOTelAttributeTranslator(attrstrans *attributes.Transla
 
 // ReceiveResourceSpans processes the given rspans and returns the source that it identified from processing them.
 func (o *OTLPReceiver) ReceiveResourceSpans(ctx context.Context, rspans ptrace.ResourceSpans, httpHeader http.Header) source.Source {
+	if o.conf.HasFeature("enable_receive_resource_spans_v2") {
+		return o.receiveResourceSpansV2(ctx, rspans, httpHeader.Get(header.ComputedStats) != "")
+	}
+	return o.receiveResourceSpansV1(ctx, rspans, httpHeader)
+}
+
+func (o *OTLPReceiver) receiveResourceSpansV2(ctx context.Context, rspans ptrace.ResourceSpans, clientComputedStats bool) source.Source {
+	otelres := rspans.Resource()
+
+	topLevelByKind := o.conf.HasFeature("enable_otlp_compute_top_level_by_span_kind")
+	tracesByID := make(map[uint64]pb.Trace)
+	priorityByID := make(map[uint64]sampler.SamplingPriority)
+	var spancount int64
+	for j := 0; j < rspans.ScopeSpans().Len(); j++ {
+		libspans := rspans.ScopeSpans().At(j)
+		for k := 0; k < libspans.Spans().Len(); k++ {
+			otelspan := libspans.Spans().At(k)
+			if _, exists := o.ignoreResNames[traceutil.GetOTelResource(otelspan, otelres)]; exists {
+				continue
+			}
+
+			spancount++
+			traceID := traceutil.OTelTraceIDToUint64(otelspan.TraceID())
+			if tracesByID[traceID] == nil {
+				tracesByID[traceID] = pb.Trace{}
+			}
+			ddspan := transform.OtelSpanToDDSpan(otelspan, otelres, libspans.Scope(), topLevelByKind, o.conf, nil)
+
+			if p, ok := ddspan.Metrics["_sampling_priority_v1"]; ok {
+				priorityByID[traceID] = sampler.SamplingPriority(p)
+			}
+			tracesByID[traceID] = append(tracesByID[traceID], ddspan)
+		}
+	}
+
+	resourceAttributes := otelres.Attributes()
+	tagstats := &info.TagStats{
+		Tags: info.Tags{
+			Lang:            traceutil.GetOTelAttrVal(resourceAttributes, true, semconv.AttributeTelemetrySDKLanguage),
```

Comment:
can we create a local variable for `traceutil.GetOTelAttrVal(resourceAttributes, true, semconv.AttributeTelemetrySDKLanguage)`